src/Utils/obiDialogParser.ts

```diff
@@ -0,0 +1,331 @@
+    
+        const dialogs: OBIDialog[] = []
+        const luMap: Map<string, string[]> = new Map()
+        const lgMap: Map<string, OBIUtils.LGItem> = new Map()
+        for (const file of files) {
+            if (file.name.endsWith('.dialog')) {
+                const fileText = await Util.readFileAsync(file)
+                const obiDialog: OBIDialog = JSON.parse(fileText)
+                // Set name, removing suffix
+                obiDialog.$id = this.removeSuffix(file.name)
+                dialogs.push(obiDialog)
+            }
+            else if (file.name.endsWith('.lu')) {
+                const fileText = await Util.readFileAsync(file)
+                this.addToLUMap(fileText, luMap)
+            }
+            else if (file.name.endsWith('.lg')) {
+                const fileText = await Util.readFileAsync(file)
+                this.addToLGMap(fileText, lgMap)
+            }
+        }
+    
+        this.composerDialog = {
+            dialogs,
+            luMap,
+            lgMap
+        }
+    
+        const mainDialog = this.composerDialog.dialogs.find(d => d.$id === "Entry.main")
+        if (!mainDialog) {
+            return null
+        }
+    
+        return this.getTrainDialogsfromOBIDialog(mainDialog)
+    }
+    
+    private addToLUMap(text: string, luMap: Map<string, string[]>): any {
+        const keys = text.split('##')
+        for (const key of keys) {
+            if (!key.startsWith(">")) {
+                const inputs = key.split('- ').map(i => i.trim())
+                luMap.set(inputs[0], inputs.slice(1))
+            }
+        }
+        return luMap
+    }
+    
+    private addToLGMap(text: string, lgMap: Map<string, OBIUtils.LGItem>): any {
+        const items = text.split('# ')
+        for (const item of items) {
+            const key = item.substring(0, item.indexOf("-")).trim()
+            const body = item.substring(item.indexOf("```") + 3, item.lastIndexOf("[")).trim()
+            const suggestionList = item.substring(item.lastIndexOf("[Suggestions=") + 13, item.lastIndexOf("]"))
+            const suggestions = suggestionList.length > 0 ? suggestionList.split('|') : []
+            const output = {
+                text: body,
+                suggestions
+            }
+            lgMap.set(key, output)
+        }
+        return lgMap
+    }
+    
+    private async getTrainDialogsfromOBIDialog(obiDialog: OBIDialog): Promise<CLM.TrainDialog[]> {
+    
+        let trainDialogs: CLM.TrainDialog[] = []
+        if (obiDialog.rules) {
+            for (const rule of obiDialog.rules) {
+                if (rule.$type === "Microsoft.IntentRule") {
+    
+                    const textVariations = this.getTextVariations(rule.intent!)
+                    const extractorStep: CLM.TrainExtractorStep = {
+                        textVariations: textVariations
+                    }
+    
+                    if (rule.steps) {
```

Comment:
This function gets pretty deeply nested.  How about putting this at the top of the for loop as `if (!rule.steps) continue` ?